vllm/distributed/utils.py

```diff
@@ -134,3 +138,29 @@ def gpu_p2p_access_check(i: int, j: int) -> bool:
         cache = json.load(f)
     _gpu_p2p_access_cache = cache
     return _gpu_p2p_access_cache[f"{i}->{j}"]
+
+
+def get_tensor_model_parallel_src_rank_and_group():
```

Comment:
Ideally this function isn't even necesssary and we can change the defaults for `broadcast_tensor_dict` to be TP group broadcast but adding this in to not break the assumption that `broadcast_tensor_dict` uses World group by default.

Reply:
We can also return a dict to use as kwargs for the `broadcast_tensor_dict` if this is less ugly:

`broadcast_tensor_dict(**get_tensor_model_parallel_src_rank_and_group())`